darts/models/forecasting/forecasting_model.py

```diff
@@ -241,6 +260,230 @@ def min_train_series_length(self) -> int:
+
+        Notes
+        -----
+        maximum target lag (second value) cannot be `None` and is always larger than 1.
+        Examples
+        --------
+        >>> model = LinearRegressionModel(lags=3, output_chunk_length=2)
+        >>> model.fit(train_series)
+        >>> model.extreme_lags
+        (-3, 2, None, None, None)
+        >>> model = LinearRegressionModel(lags=[3, 5], past_covariates_lags = 4, output_chunk_length=7)
+        >>> model.fit(train_series, past_covariates=past_covariates)
+        >>> model.extreme_lags
+        (-5, 7, -4, None, None)
+        >>> model = LinearRegressionModel(lags=[3, 5], future_covariates_lags = [4, 6], output_chunk_length=7)
+        >>> model.fit(train_series, future_covariates=future_covariates)
+        >>> model.extreme_lags
+        (-5, 7, None, 4, 6)
+        >>> model = NBEATSModel(input_chunk_length=10, output_chunk_length=7)
+        >>> model.fit(train_series)
+        >>> model.extreme_lags
+        (-10, 7, None, None, None)
+        >>> model = NBEATSModel(input_chunk_length=10, output_chunk_length=7, future_covariates_lags=[4, 6])
+        >>> model.fit(train_series, future_covariates)
+        >>> model.extreme_lags
+        (-10, 7, None, 4, 6)
+        """
+
+        return (-1, 1, None, None, None)
+
+    @property
+    def training_sample_time_index_length(self) -> int:
+        """
+        Class property defining the required time_index length for one training sample, for any model.
+        """
+        (
+            min_target_lag,
+            max_target_lag,
+            min_past_cov_lag,
+            min_future_cov_lag,
+            max_future_cov_lag,
+        ) = self.extreme_lags
+
+        return max(
+            max_target_lag,
+            max_future_cov_lag if max_future_cov_lag else 0,
+        ) - min(
+            min_target_lag if min_target_lag else 0,
+            min_past_cov_lag if min_past_cov_lag else 0,
+            min_future_cov_lag if min_future_cov_lag else 0,
+        )
+
+    @property
+    def predict_sample_time_index_length(self) -> int:
+        """
+        Class property defining the required time_index length for one predict sample, for any model.
+        A predict sample is the minimum required set of series and covariates chunks to be able to predict
+        a single point.
+        """
+        (
+            min_target_lag,
+            max_target_lag,
+            min_past_cov_lag,
+            min_future_cov_lag,
+            max_future_cov_lag,
+        ) = self.extreme_lags
+
+        return (
+            max_future_cov_lag
+            if max_future_cov_lag
+            else 0
+            - min(
+                min_target_lag if min_target_lag else 0,
+                min_past_cov_lag if min_past_cov_lag else 0,
+                min_future_cov_lag if min_future_cov_lag else 0,
+            )
+        )
+
+    def _get_historical_forecastable_time_index(
```

Comment:
If this is called only from `historical_forecasts()`, how about moving it inside the `historical_forecasts()` function?